Source/CarthageKit/Cartfile.swift

```diff
@@ -165,3 +166,43 @@ extension ResolvedCartfile: CustomStringConvertible {
 			.joined(separator: "")
 	}
 }
+
+
+extension String {
+	
+	/// Returns self without any potential trailing Cartfile comment. A Cartfile
+	/// comment starts with the first `commentIndicator` that is not embedded in any quote
+	var strippingTrailingCartfileComment: String {
+		
+		// since Cartfile syntax doesn't support nested quotes, such as `"version-\"alpha\""`
+		// I can just consider any odd-number occurence of a quote as a quote-start, and any
+		// even-numbered occurrence of a quote as quote-end.
+		// The comment indicator (e.g. `#`) is the start of a comment if it's not nested in quotes.
+		// The following code works also for comment indicators that are are more than one character long (e.g. "//")
+		
+		let quote = "\""
+		
+		// Splitting the string by quote will make odd-numbered chunks outside of quotes, and
+		// even-numbered chunks inside of quotes.
+		// `omittingEmptySubsequences` is needed to maintain this property even in case of empty quotes.
+		let quoteDelimitedChunks = self.split(separator: quote.first!,
+											  maxSplits: Int.max,
+											  omittingEmptySubsequences: false)
+		
+		
+		for (offset, chunk) in quoteDelimitedChunks.enumerated() {
+			let isInQuote = offset % 2 == 1 // even chunks are not in quotes, see comment above
+			if isInQuote {
+				continue // we don't consider comment indicators inside quotes
+			}
+			if let range = chunk.range(of: Cartfile.commentIndicator) {
+				// there is a comment, return everything before its position
+				let previousChunks = quoteDelimitedChunks[..<offset]
```

Comment:
Even though this code is well-considered as it stands — in that we could assert `startIndex` for `quoteDelimitedChunks` always being `0` based on the singular way we initialize it — just to prevent any potential of breaking on someone else’s refactor, let’s:

```swift
				let advancedOffset = (..<offset).relative(to: quoteDelimitedChunks)
				let previousChunks = quoteDelimitedChunks[advancedOffset]
```